crates/miden-testing/src/mock_chain/chain.rs

```diff
@@ -599,35 +623,48 @@ impl MockChain {
             },
         };
 
-        let tx_inputs = self
-            .get_transaction_inputs(
-                mock_account.account().clone(),
-                mock_account.seed().cloned(),
-                note_ids,
-                unauthenticated_notes,
-            )
-            .context("failed to gather transaction inputs")?;
+        let tx_inputs = self.get_transaction_inputs_at_block(
+            reference_block,
+            mock_account.account().clone(),
+            mock_account.seed().cloned(),
+            note_ids,
+            unauthenticated_notes,
+        )?;
 
-        let tx_context_builder = TransactionContextBuilder::new(mock_account.account().clone())
+        Ok(TransactionContextBuilder::new(mock_account.account().clone())
             .authenticator(mock_account.authenticator().cloned())
             .account_seed(mock_account.seed().cloned())
-            .tx_inputs(tx_inputs);
+            .tx_inputs(tx_inputs))
+    }
 
-        Ok(tx_context_builder)
+    /// Initializes a [`TransactionContextBuilder`] for executing against the last block header.
+    ///
+    /// Please refer to the docs for [`Self::build_tx_context_at_block`] for a complete explanation
+    /// of the function.
+    pub fn build_tx_context(
```

Comment:
Nit: I'd maybe write something to the effect of "This is a wrapper around `build_tx_context_at` which uses the latest block as the reference block. See that function's docs for details".